pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -374,6 +385,59 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	return nil
 }
 
+// newExternalCephObjectStoreInstances returns a set of CephObjectStores
+// needed for external cluster mode
+func (r *ReconcileStorageCluster) newExternalCephObjectStoreInstances(
+	initData *ocsv1.StorageCluster, rgwEndpoint string, reqLogger logr.Logger) ([]*cephv1.CephObjectStore, error) {
+	// check whether the provided rgw endpoint is empty
+	if rgwEndpoint = strings.TrimSpace(rgwEndpoint); rgwEndpoint == "" {
+		reqLogger.Info("WARNING: Empty RGW Endpoint specified, external CephObjectStore won't be created")
+		return nil, nil
+	}
+
+	var gateWay cephv1.GatewaySpec
+	hostIP, portStr, err := net.SplitHostPort(rgwEndpoint)
+	if err != nil {
+		reqLogger.Error(err,
+			fmt.Sprintf("invalid rgw endpoint provided: %s", rgwEndpoint))
+		return nil, err
+	}
+	if hostIP == "" {
+		err := fmt.Errorf("An empty rgw host 'IP' address found")
+		reqLogger.Error(err, err.Error())
```

Comment:
Why not just `reqLogger.Error(err)`?

Reply:
`reqLogger.Error` requires a second string argument, which provides a message. Will change to a different message (rather than the same error message).